src/rules/no-reversed-media-syntax.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * @fileoverview Rule to prevent reversed link and image syntax in Markdown.
+ * @author xbinaryx
+ */
+
+//-----------------------------------------------------------------------------
+// Imports
+//-----------------------------------------------------------------------------
+
+import { findOffsets } from "../util.js";
+
+//-----------------------------------------------------------------------------
+// Type Definitions
+//-----------------------------------------------------------------------------
+
+/**
+ * @typedef {import("../types.ts").MarkdownRuleDefinition<{ RuleOptions: []; }>}
+ * NoReversedMediaSyntaxRuleDefinition
+ */
+
+//-----------------------------------------------------------------------------
+// Helpers
+//-----------------------------------------------------------------------------
+
+/** Matches reversed link/image syntax like (text)[url], ignoring escaped characters like \(text\)[url]. */
+const reversedPattern =
+	/(?<!\\)\(((?:\\.|[^()\\]|\([\s\S]*\))*)\)\[((?:\\.|[^\]\\\n])*)\](?!\()/gu;
+const codeSpanPattern = /(?<!\\)`+[^`]*`+/gu;
+
+/**
+ * Checks if a match is within any of the code spans
+ * @param {number} matchIndex The index of the match
+ * @param {Array<{start: number, end: number}>} codeSpans Array of code span positions
+ * @returns {boolean} True if the match is within a code span
+ */
+function isInCodeSpan(matchIndex, codeSpans) {
+	return codeSpans.some(
+		span => matchIndex >= span.start && matchIndex < span.end,
+	);
+}
+
+/**
+ * Finds all code spans in the text
+ * @param {string} text The text to search
+ * @returns {Array<{start: number, end: number}>} Array of code span positions
+ */
+function findCodeSpans(text) {
+	const codeSpans = [];
+	let match;
+
+	while ((match = codeSpanPattern.exec(text)) !== null) {
+		codeSpans.push({
+			start: match.index,
+			end: match.index + match[0].length,
+		});
+	}
+
+	return codeSpans;
+}
+
+//-----------------------------------------------------------------------------
+// Rule Definition
+//-----------------------------------------------------------------------------
+
+/** @type {NoReversedMediaSyntaxRuleDefinition} */
+export default {
+	meta: {
+		type: "problem",
+
+		docs: {
+			recommended: true,
+			description: "Disallow reversed link and image syntax",
+			url: "https://github.com/eslint/markdown/blob/main/docs/rules/no-reversed-media-syntax.md",
+		},
+
+		fixable: "code",
+
+		messages: {
+			reversedSyntax:
+				"Unexpected reversed syntax found. Use [{{label}}]({{url}}) instead.",
```

Comment:
Some URLs are very long i think that would make this error message too long, we can try this:
`"Unexpected reversed syntax found. Use [label](URL) syntax instead."`.

Also, for both the link `[]()` and image syntax `![]()`, we are suggesting the same syntax in message `[]()`. I feel we can suggest their respective syntax?


What do you think?

Reply:
Makes sense to me.